libraries/libloader/src/main/java/org/pentaho/reporting/libraries/resourceloader/loader/URLResourceLoader.java

```diff
@@ -70,13 +70,22 @@ public ResourceKey createKey( final Object value, final Map factoryKeys ) throws
       return new ResourceKey( SCHEMA_NAME, value, factoryKeys );
     }
     if ( value instanceof String ) {
-      final String valueString = (String) value;
+      String valueString = (String) value;
+      final String encodedValue;
+      // query parameters might contain ' ' (space) that must be escaped
+      // other 'special' characters shouldn't need escaping, as they are
+      // relaxedQueryChars in Tomcat config
+      if ( valueString.contains( " " ) ) {
```

Comment:
Why not replace this if else just by: encodedValue = valueString.replace( " ", "%20" );

If you do not have chars to replace, you do not do the replacement. In terms of performance, it is very similar because you do not have to call contains.

Reply:
Upon further investigation, it was found that, indeed, the contains provided no benefit and, as suggested, the validation was removed